src/angular-app/bellows/shared/audio-recorder/audio-recorder.component.ts

```diff
@@ -1,163 +1,126 @@
-import * as lamejs from 'lamejs';
 
-declare var webkitAudioContext: {
-  new(): AudioContext;
-};
+declare var MediaRecorder: any;
 
 export class AudioRecorderController implements angular.IController {
 
   static $inject = ['$interval', '$scope'];
-  constructor(private $interval: angular.IIntervalService, private $scope: angular.IScope) {}
 
+  mediaRecorder: any;
+  chunks: any = [];
   isRecording = false;
   hasRecorded = false;
+  recordingStartTime: Date;
   audioSrc: string;
-  blob: Blob;
+  blob: any;
   recordingTime: string;
   errorMessage: string;
-  stopMediaStream: () => void;
   callback: (blob: Blob) => void;
+  durationInMilliseconds: number;
   interval: angular.IPromise<void>;
 
-  toggleRecording() {
-    if (this.isRecording) this.stopRecording();
-    else this.startRecording();
-    this.isRecording = !this.isRecording;
-  }
-
-  close() {
-    this.stopRecording();
-    this.callback(null);
-  }
-
-  saveAudio() {
-    this.callback(this.blob);
-  }
-
-  recordingSupported() {
-    return navigator.mediaDevices && navigator.mediaDevices.enumerateDevices && navigator.mediaDevices.getUserMedia &&
-      ((window as any).AudioContext || (window as any).webkitAudioContext);
-  }
+  constructor(private $interval: angular.IIntervalService, private $scope: angular.IScope) {}
 
-  $onDestroy() {
-    this.stopRecording();
-  }
+//METHODS//
 
   private startRecording() {
+
     this.recordingTime = '0:00';
 
-    navigator.mediaDevices.getUserMedia({audio: true}).then(stream => {
+    //getUserMedia
+    navigator.mediaDevices.getUserMedia({audio: true})
+    .then(stream => {
+
+      this.mediaRecorder = new MediaRecorder(stream);
 
       this.$scope.$apply(() => {
         this.hasRecorded = true;
         this.errorMessage = null;
         this.isRecording = true;
       });
 
-      const recordingStartTime = new Date();
-      // webkit prefix required for Safari
-      const context = (window as any).AudioContext ? new AudioContext() : new webkitAudioContext();
-      const bufferSize = 0;
-      const channels = 1;
-      const processor = context.createScriptProcessor(bufferSize, channels, channels);
-      context.createMediaStreamSource(stream).connect(processor);
-      processor.connect(context.destination);
-      const sampleRate = context.sampleRate;
-      const bitrate = 128;
-      const mp3Encoder = new MP3Encoder(channels, sampleRate, bitrate);
-
-      function handleAudioData(event: AudioProcessingEvent) {
-        mp3Encoder.appendData(event.inputBuffer.getChannelData(0));
-      }
-      processor.addEventListener('audioprocess', handleAudioData);
-
-      mp3Encoder.onMp3Blob(blob => {
-        this.blob = blob;
-        this.audioSrc = URL.createObjectURL(blob);
+
+      this.mediaRecorder.addEventListener('dataavailable', async (e: { data: any; }) => {
+        this.chunks.push(e.data);
+        var roughBlob = new Blob(this.chunks, {type: 'audio/webm; codecs=opus'});
+        //2022-09
```

Comment:
nit: this date comment can come on the same line as the next comment

Reply:
Thanks, now edited